### PR TITLE
OAK-10250: Configurable suspend timeout

### PR DESCRIPTION
**Pull request by mreutegg**

### CODE REVIEW

File: oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/Configuration.java [hunk 1]
```diff
@@ -281,4 +282,10 @@
             name = "Document Node Store Compression",
             description = "Select compressor type for collections. 'Snappy' is the default supported compression.")
     String collectionCompressionType() default "snappy";
+
+    @AttributeDefinition(
+            name = "Suspend timeout",
+            description = "Timeout for a suspended commit after it conflicted with" +
+                    "a change that is not yet visible. Default: 60000 (milliseconds).")
```
Comment (stefan-egli) [inline oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/Configuration.java]: this hardcodes the default to 60000 in osgi metadata - could maybe also refer to DEFAULT_SUSPEND_TIMEOUT?

**Review by stefan-egli — APPROVED** [review]
one minor comment, +1 otherwise

File: oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/Configuration.java [hunk 2]
```diff
@@ -281,4 +282,10 @@
...
+    long suspendTimeoutMillis() default DEFAULT_SUSPEND_TIMEOUT;
```
Comment (rishabhdaim) [inline oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/Configuration.java:291]:
```suggestion
    long commitSuspendTimeout() default DEFAULT_SUSPEND_TIMEOUT;
```

File: oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/Configuration.java [hunk 3]
```diff
@@ -281,4 +282,10 @@
             name = "Document Node Store Compression",
             description = "Select compressor type for collections. 'Snappy' is the default supported compression.")
     String collectionCompressionType() default "snappy";
+
+    @AttributeDefinition(
+            name = "Suspend timeout",
```
Comment (rishabhdaim) [inline oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/Configuration.java]:
```suggestion
            name = "Commit Suspend timeout",
```

File: oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/Configuration.java (hunk 2 above)
Comment (rishabhdaim) [reply oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/Configuration.java:291]: I would avoid `millis` in API name. This is standard across all the other APIs

**Review by rishabhdaim — CHANGES_REQUESTED** [review]

File: oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/Configuration.java (hunk 2 above)
Comment (mreutegg) [reply oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/Configuration.java:291]: We actually have other configuration methods with time units in the name:
- `blobGcMaxAgeInSecs()`
- `blobTrackSnapshotIntervalInSecs()`
- `maxReplicationLagInSecs()`

And I also wanted to keep the name in line with the system property that is already there: `oak.documentMK.suspendTimeoutMillis`

File: oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/Configuration.java (hunk 3 above)
Comment (mreutegg) [reply oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/Configuration.java]: I'll make that change in my next commit.

**Review by Joscorbe — APPROVED** [review]
+1

File: oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/Configuration.java (hunk 2 above)
Comment (rishabhdaim) [reply oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/Configuration.java:291]: I missed the properties in `Secs`

**Review by rishabhdaim — APPROVED** [review]

**Comment by mreutegg** [comment]
Tests in oak-core run out of memory. See [OAK-10259](https://issues.apache.org/jira/browse/OAK-10259). oak-run and oak-upgrade failed to start docker container. The failure in oak-jcr I'm not able to reproduce locally.

**Pull request merged**